azure/infrastructure.tf

```diff
@@ -15,13 +16,13 @@ resource "azurerm_virtual_network" "virtualNetwork" {
   name                = "${var.cluster_name}_vnet"
   address_space       = ["10.0.0.0/16"]
   location            = var.location
-  resource_group_name = azurerm_resource_group.group.name
+  resource_group_name = length(var.azure_resource_group) == 0 ? azurerm_resource_group.group[0].name : var.azure_resource_group
```

Comment:
Instead of checking the length of `var.azure_resource_group` each time we need the resource group name, we can create a local variable that will contain either the result of the comparison:
```
locals {
  resource_group_name = var.azure_resource_group == "" ? azurerm_resource_group.group[0].name : var.azure_resource_group
}
```
Then we can use `local.azure_resource_group`.